subnet.go

```diff
@@ -35,7 +37,9 @@ type subnet struct {
 // SubnetArgs is an argument struct used to create a
 // new internal subnet type that supports the Subnet interface.
 type SubnetArgs struct {
+	// Deprecated in favor of UUID
 	ID                string
```

Comment:
Since this is v6, we can get rid of ID since it's being replaced bu UUID.

Reply:
So we discussed with @jameinel and @SimonRichardson and the idea is to keep these deprecated fields to ensure not breaking in the future (although this version shouldn't be used in juju <4). This is coherent to what was done previously in subnets with `SpaceName`.

Reply:
Yeah, we will need to reconcile V3 to V4 by the IDs anyway.